util/types/src/core/cell.rs

```diff
@@ -424,16 +446,27 @@ pub fn resolve_transaction<CP: CellProvider, HC: HeaderChecker, S: BuildHasher>(
         }
     }
 
+    let system_cell = SYSTEM_CELL.get().expect("SYSTEM_CELL is not initialized");
```

Comment:
I think a more elegant solution is creating a wrapper around the resolver, like:

```
struct<F> CellProviderWithSystemCellCache {
  system_cell: SystemCell,
  inner: F,
}
```

Reply:
This PR is a draft for bench, I will improve it later.
But I think `CellProvider` is also a bad idea, this cache is for dep, not input.